lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -398,24 +349,65 @@ void WebsocketTlsTPM::client_loop() {
 
     info.fd_limit_per_thread = 1 + 1 + 1;
 
-    bool use_tpm = connection_options.use_tpm_tls;
+    // Setup context - need to know the key type first
+
+    std::string path_key;
+    std::string path_chain;
+    std::optional<std::string> password;
 
-    // Setup context
-    OSSL_LIB_CTX* lib_ctx;
-    SSL_CTX* ssl_ctx;
+    if (this->connection_options.security_profile == 3) {
 
-    create_sec_context(use_tpm, lib_ctx, ssl_ctx);
+        const auto certificate_key_pair =
+            this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
 
-    // Connection acquire the contexts
-    conn_data->sec_lib_context = std::unique_ptr<OSSL_LIB_CTX>(lib_ctx);
-    conn_data->sec_context = std::unique_ptr<SSL_CTX>(ssl_ctx);
+        if (!certificate_key_pair.has_value()) {
+            EVLOG_AND_THROW(std::runtime_error(
+                "Connecting with security profile 3 but no client side certificate is present or valid"));
+        }
+
+        path_chain = std::string(certificate_key_pair.value().certificate_path.c_str());
+        path_key = std::string(certificate_key_pair.value().key_path.c_str());
+        password = certificate_key_pair.value().password;
+    }
+
+    SSL_CTX* ssl_ctx = nullptr;
+    bool tpm_key = false;
+
+    {
+        OpenSSLProvider provider;
```

Comment:
Question: the propquery, who sets it here?

Reply:
it's done a few lines later

Reply:
No, I meant who sets: 

static const char* s_default_provider = PROPQUERY_DEFAULT;
static const char* s_tpm2_provider = PROPQUERY_TPM2;

I know in libevse it's done in the CMAKE. Are those defaults used? Are they configurable for libocpp too? Or they're common? How is that handled when we require TPM with libevse AND libocpp?

Reply:
The property strings are only in libevse-security. Users of libevse-security chose the settings to apply via OpenSSLProvider::mode_t. This deliberately prevents setting other property strings and causing runtime issues.
Hence libocpp just needs to decide to use mode_t::default_provider or mode_t::tpm2_provider
The coder doesn't need to know how that happens (within libevse-security)